### PR TITLE
Fix map hexlayer refresh

### PR DESCRIPTION
**Pull request by karimkawambwa**
Fixes issue where the hexlayer sticks around after map location is changed on the url hash.

### CODE REVIEW

**Comment by karimkawambwa** [comment]
cc @kilemensi i separated it here.

File: src/utils.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+export function throttle (callback, wait, immediate = false) {
```
Comment (kilemensi) [inline src/utils.js]: I see the project already uses `lodash` & `lodash` has `throttle` [method](https://lodash.com/docs#throttle). Can you add a comment why we need a custom one?

File: src/map.vue [hunk 2]
```diff
@@ -2,65 +2,96 @@
...
+    const refresh = throttle(() => {
+        map.removeLayer(this.hexLayer);
+        this.hexLayer.data([]);
+        this.hexLayer.addTo(map);
+        this.hexLayer.data(this.data);
+    }, 200); // hash has a default throttle of 100 ms
 
-      let hexLayer = new leaflet.HexbinLayer(options).addTo(map)
+    map.on('dragend', () => {
+      map.dragged = true; // don't reload when dragged
```
Comment (kilemensi) [inline src/map.vue]: QQ: Does this ever get reset back to `false`? Or once you've dragged once, it will always be `true`

**Review by kilemensi — CHANGES_REQUESTED** [review]
Thanks @karimkawambwa ... I can at least see what you were talking about now.

File: src/map.vue (hunk 2 above)
Comment (karimkawambwa) [reply src/map.vue]: Yeah it gets reset https://github.com/CodeForAfrica/sensors.AFRICA-AQ-map/pull/17/files#diff-10329629e4e003e8671762aa141434b8R85

File: src/utils.js (hunk 1 above)
Comment (karimkawambwa) [reply src/utils.js]: I used it but i did not see the previous call being cancelled so i did not want to fight another framework.

File: src/map.vue (hunk 2 above)
Comment (kilemensi) [reply src/map.vue]: Damn, [return-assign](https://eslint.org/docs/rules/no-return-assign)

Comment (karimkawambwa) [reply src/map.vue]: @kilemensi resolved

File: src/utils.js (hunk 1 above)
Comment (kilemensi) [reply src/utils.js]: I'm not sure what you mean by previous calls being cancelled @karimkawambwa... If you want to cancel I think you have to do that manually, `_.throttle` just ensures the calls are spaced out in time. If you want the other calls made within a given time to automatically be dropped, I think `_.debounce` is what you need.

Comment (karimkawambwa) [reply src/utils.js]: Cool i want `_.debounce` behavior

**Comment by karimkawambwa** [comment]
@kilemensi resolved sir

**Review by kilemensi — APPROVED** [review]
:+1: @karimkawambwa ... Over to you.

**Pull request merged**